docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,287 @@
+2. Retrieve the symmetric key `enc(SYM_KEY)` from the server.
+3. Decrypt the key with `PRIV_AUTH_KEY`.
+4. Encrypt the symmetric key with `PUB_AUTH_KEY_B` (`enc'(SYM_KEY)`).
+5. Upload `enc'(SYM_KEY)`, `PUB_AUTH_KEY_B` and the secret of the new authentication method to the server.
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as Authentification server
+
+  Note over Alice,S: Alice is already authenticated to the server
+
+  Alice ->> Alice: Create a new asymmetric key pair (`PRIV_AUTH_KEY_B`/`PUB_AUTH_KEY_B`)
+  S ->> Alice: Retrieve the key `enc(SYM_KEY)` related to `PUB_AUTH_KEY`
+  Alice ->> Alice: Decrypt the key `enc(SYM_KEY)` with `PRIV_AUTH_KEY`
+  Alice ->> Alice: Encrypt the key `SYM_KEY` with `PUB_AUTH_KEY_B` (`enc'(SYM_KEY)`)
+  Alice ->> S: Upload `enc'(SYM_KEY)`, `PUB_AUTH_KEY_B` and the secret of the authentification method
+```
+
+### Removing an authentication method
+
+To remove an authentication method from a user, it requires the user to:
+
+1. Retrieve all `PUB_AUTH_KEY` except the one to remove.
+2. Retrieve `enc(SYM_KEY)` related to the current authentication method.
+3. Get all devices `ENC_DEV`.
+4. Decrypt `enc(SYM_KEY)` with `PRIV_AUTH_KEY`.
+5. Create a new symmetric key `SYM_KEY_B`.
+6. For each device:
+
+   1. Decrypt the device with `SYM_KEY`.
+   2. Encrypt the device with `SYM_KEY_B`.
+
+7. For each `PUB_AUTH_KEY`, encrypt `SYM_KEY` with it.
+8. Upload new list of encrypted devices, the list of `enc(SYM_KEY_B)` of each `PUB_AUTH_KEY`
+9. Remove the authentication method from the server.
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as Authentification server
+
+  Note over Alice,S: Alice is already authenticated to the server
+
+  S ->> Alice: Retrieve every key `PUB_AUTH_KEY` except the one to remove
+  S ->> Alice: Get the symmetric key `enc(SYM_KEY)`
+  S ->> Alice: Download every devices `ENC_DEV`
+  Alice ->> Alice: Decrypt `enc(SYM_KEY)` with `PRIV_AUTH_KEY`
+  Alice ->> Alice: Create a new symmetric key (`SYM_KEY_B`)
+  loop For each device
+    Alice ->> Alice: Decrypt the device with `SYM_KEY`
+    Alice ->> Alice: Encrypt the device with `SYM_KEY_B` (`ENC_DEV_B`)
+  end
+  loop For each `PUB_AUTH_KEY`
+    Alice ->> Alice: Encrypt the key `SYM_KEY_B` with `PUB_AUTH_KEY` (`enc(SYM_KEY_B)`)
+  end
+  Alice ->> S: Upload the list of encrypted device `ENC_DEV_B`
+  Alice ->> S: Upload the list of key `enc(SYM_KEY_B)`
+  Alice ->> S: Remove the authentication method from the server
+```
+
+### Using the service alongside the parsec-cloud server
+
+The parsec client would use the authentication service to store a special device, who will only be used to create a new local device.
+
+That new device would be stored on the local storage and if the storage happens to be cleaned, the user could still re-create a new device from the special one stored on the service.
```

Comment:
This section needs further details. We should explain a little bit why this "special device" and local devices are needed, why not directly using the special device to login, and _how_ exactly the special device is used to "create a new local device".

Also, we should probably give them more meaningful names. I think we've already called them "device registration key" (the one that is uploaded and allows to register new devices) and "device security key" (the "classic" Parsec device key).

Reply:
I've added more details why the device is not used as is